docs/v3tov4.md

```diff
@@ -0,0 +1,340 @@
+
+Description of the paramaters:
+
+`timestamp` (time.Time) is the metric's start time in Unix time (milliseconds).
+
+`name` (string) is the name of the metric. The value must be less than 255 characters.
+
+`value` (float64) is a double. Gauges values can be any value positive or negative. Cumulative-rate values need to be 
+ever-growing.
+
+`count` (float64) is the number of occurrences of an event reported in a given interval. Must be a positive value.
+
+`average` (float64) is the number expressing the central value in the set of data.
+
+`sum` (float64) is the aggregation of all the values registered in the set of data.
+
+`min` (float64) is the minimum value registered in the set of data.
+
+`max` (float64) is the maximum value registered in the set of data.
+
+## Mapping metrics types from v3
+
+* GAUGE / RATE
+
+  No changes here. Both types work in the same way as before.
+
+* PRATE
+
+  It maps to the new `cumulative-rate` metric.
+
+* DELTA / PDELTA
+
+  They map to the new `cumulative-count` metric.
+  
+* ATTRIBUTES
+
+  This metric type has been removed. Now every metric can have attributes, called dimensions, attached to it.
+
+#### Summary
+
+| type           | v3 | v4 | mapping                                                                       |
+|----------------|----|----|-------------------------------------------------------------------------------|
+| gauge          | ✅ | ✅ | integration.Gauge(timestamp, name, value)                                     |
+| rate           | ✅ | ✅ | integration.Rate(timestamp, name, value)                                      |
+| prate          | ✅ | ❌ | integration.CumulativeRate(timestamp, name, value)                            |
+| delta / pdelta | ✅ | ❌ | integration.CumulativeCount(timestamp, name, count)                           |
+| attribute      | ✅ | ❌ | add dimensions to the metrics                                                 |
+| count          | ❌ | ✅ | integration.Count(timestamp, interval, name, count)                           |
+| summary        | ❌ | ✅ | integration.Summary(timestamp, interval, name, count, average, sum, min, max) |
+
+#### Adding dimensions to a metric
+
+Metrics can have a list of dimensions. These dimensions, along with the name, make the metric unique and can be used to 
+search and filter them.
+
+To add a new Dimensions just call on a metric 
+
+`func (i *<Gauge|Count|Summary>) AddDimension(key string, value interface{})`
+
+If the `key` already exists the previous value will be overwritten with the new one.
+
+#### Adding metrics to an Entity
+
+Once you have a metric, it can be attached to an Entity. The metric will belong to that Entity and it will be decorated
+with the Entity's tags.
+
+To add a metric to an entity just call the method on an Entity
+
+`func (e *Entity) AddMetric(metric)`
+
+### Events
+
+The API for creating events has changed a little bit. Events can be attached to an Entity or Integration.
```

Comment:
"or Integration" ? Maybe you mean HostEntity?

Reply:
good catch